rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -644,57 +644,64 @@ def find_recursive_deployments(self, source_id):
                         visited[dependency.target_deployment_id] = True
         return results
 
-    def increase_deployment_counts_in_graph(self, target, source):
+    def increase_deployment_counts_in_graph(self, target_id, source):
         """
         Increase the deployment counts for target deployment that is
         referenced directly by the `source` deployment and then make sure to
         propagate that increase to all deployment that are referenced
         directly and indirectly by `target` deployment
-        :param target: Target Deployment node that we need to attach source to
-        :rtype Deployment
+        :param target_id: Target Deployment ID that we need to attach
```

Comment:
I think it's better to continue to  pass the target deployment in here rather than the ID itself- we recently changed to this approach in several places.

Reply:
@geokala actually there is a reason for that which is to lock that deployment https://github.com/cloudify-cosmo/cloudify-manager/blob/RD-1485-update-deployment-apis/rest-service/manager_rest/rest/rest_utils.py#L669 here thats why `id` is passed instead of the object because I want to update the deployment counts and use `locking`

Reply:
OK, makes sense